docs/infra/treatment-service.md

```diff
@@ -0,0 +1,132 @@
+
+## Overview
+XP's Treatment Service is currently offered in two different versions:
+
+- a central Treatment Service, *deployed as a single deployment, to serve treatments to multiple 
+  clients/users with multiple projects*
+- a Treatment Service plugin, *deployed as a Turing Router experiment engine plugin, to serve treatments to a single 
+  client/user with a single project*
+
+## Central Treatment Service
+
+### Requirements
+- Configuration
+- Google Cloud Provider (GCP) service account
+
+### Configuration
+The central Treatment Service deployment uses a variety of configuration that can be stored in one or multiple 
+configuration files (`.yaml`) files. Some of such examples can be found here:
+
+- https://github.com/caraml-dev/xp/treatment-service/config/example.yaml
+- https://github.com/caraml-dev/xp/treatment-service/testdata/config1.yaml
+- https://github.com/caraml-dev/xp/treatment-service/testdata/config2.yaml
+
+As can be seen from the various examples, there are a number of values that need to be set, though not all of them 
+are necessarily required. Some of these values, when left undefined/empty, will be automatically initialised with 
+certain default values (see https://github.com/caraml-dev/xp/blob/main/treatment-service/config/config.go#L22).
+
+### Google Cloud Provider (GCP) Service Account
+[Google Cloud Pub/Sub](https://cloud.google.com/pubsub/docs/overview) is used to allow the Treatment Service to 
+communicate with the Management Service to retrieve information about the experiments that are being run at any point 
+of time. More specifically, the Treatment Service subscribes to a Pub/Sub topic that the Management Service 
+publishes to whenever there are updates to actively running experiments.
+
+Furthermore, the Treatment Service may also interact with [Google BigQuery](https://cloud.google.com/bigquery), if 
+it has been set up, to perform logging of all the treatments it assigns in a specified table.
+
+Hence, if access control has been set up for the Pub/Sub topic or for the BigQuery table, the Treatment Service needs 
+to be authenticated using a service account key that has the necessary rights to subscribe to the aforementioned Pub/Sub 
+topic or to push data to the said BigQuery table. It does so by accessing a `.json` 
+[file containing the service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys), 
+whose location is stored as a filepath in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable. 
+
+Depending on how the Treatment Service is deployed, there are a variety of ways to ensure that it has access to 
+the service account key file and the environment variable. See the section below for more information.
+
+Note that the Treatment Service does not currently support the usage of multiple GCP service accounts for 
+authentication, i.e the Pub/Sub topic and the BigQuery table each requiring a **different** service account for 
+authentication.
```

Comment:
For ease of understanding, could you also list down the minimal roles / permissions required on this account, like:

* roles/pubsub.subscriber
* roles/bigquery.dataEditor
* Some permission for a dataset specifically?

(Not sure if these are the exact roles needed, I just added an example.)

Reply:
Okay, I just added this paragraph below to describe the minimum roles required: 

```markdown
In short, the service account configured would minimally need these roles/permissions:
- [roles/pubsub.subscriber](https://cloud.google.com/pubsub/docs/access-control#roles), minimally, for the topic 
  that the Management Service will publish updates to
- [roles/bigquery.dataEditor](https://cloud.google.com/bigquery/docs/access-control#bigquery.dataEditor), minimally, 
  for the table that will contain the logs (if logging to BigQuery is configured)
```